lldb/source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp

```diff
@@ -1009,6 +1009,117 @@ GetPatternBindingForVarDecl(swift::VarDecl *var_decl,
+  CompilerType referent_type =
+      type_system_swift->GetReferentType(variable.m_type.GetOpaqueQualType());
+
+  swift::Type swift_type = GetSwiftType(referent_type);
+  if (!swift_type)
+    return {};
+
+  // One tricky bit here is that this var may be an argument to the
+  // function whose context we are emulating, and that argument might be
+  // of "inout" type.  We need to strip the inout off the type or the
+  // initial parse will fail.  Fortunately, the variable access goes the
+  // same regardless of whether it is inout or not, so we don't have to do
+  // anything more to get this to work.
+  swift_type = swift_type->getWithoutSpecifierType();
+
+  if (is_self) {
+    // Another tricky bit is that the Metatype types we get have the
+    // "Representation" already attached (i.e.
+    // "@thick", "@thin".)  But the representation is a SIL level thing,
+    // and if it is attached to types that we hand the parser, it throws a
+    // verifier error & aborts.  So we strip it off here:
+    swift::MetatypeType *metatype_type =
+        llvm::dyn_cast<swift::MetatypeType>(swift_type.getPointer());
+    if (metatype_type) {
+      swift_type = swift::Type(
+          swift::MetatypeType::get(metatype_type->getInstanceType()));
+    }
+  }
+
+  if (swift_type->hasArchetype())
+    swift_type = swift_type->mapTypeOutOfContext();
+
+  return {swift_type};
+}
+
+swift::VarDecl *SwiftASTManipulator::GetVarDeclForVariableInFunction(
+    const SwiftASTManipulator::VariableInfo &variable, bool is_self,
+    swift::FuncDecl *containing_function) {
+  const auto maybe_swift_type = GetSwiftTypeForVariable(variable, is_self);
+
+  if (!maybe_swift_type)
+    return {};
+
+  const auto &swift_type = *maybe_swift_type;
+
+  const swift::SourceLoc loc = containing_function->getBody()->getLBraceLoc();
+  const swift::Identifier name = variable.GetName();
+  // We may need to mutate the self variable later, so hardcode it to a var
+  // in that case.
+  const auto introducer =
+      is_self ? swift::VarDecl::Introducer::Var : variable.GetVarIntroducer();
+
+  const swift::ASTContext &ast_context = m_source_file.getASTContext();
+  swift::VarDecl *redirected_var_decl = new (ast_context) swift::VarDecl(
+      /*is_staic*/ false, introducer, loc, name, containing_function);
+
+  redirected_var_decl->setInterfaceType(swift_type);
+  redirected_var_decl->setDebuggerVar(true);
+  redirected_var_decl->setImplicit(true);
+
+  // This avoids having local variables filtered out by
+  // swift::namelookup::filterForDiscriminator().
+  redirected_var_decl->overwriteAccess(swift::AccessLevel::Public);
+
+  if (swift_type->getAs<swift::WeakStorageType>()) {
+    redirected_var_decl->getAttrs().add(
+        new (ast_context) swift::ReferenceOwnershipAttr(
+            swift::SourceRange(), swift::ReferenceOwnership::Weak));
+  }
+
+  return redirected_var_decl;
+}
+
+/// Adds the new nodes to the beginning of the function.
```

Comment:
The comment here basically repeats the function name with more words. Maybe we can explain what's expected to be in new_nodes?
Also, should this be an ArrayRef instead of a reference to a SmallVector?